src/helpers/GoalController.ts

```diff
@@ -62,7 +71,19 @@ export const modifyGoal = async (
   goalTitle: string,
   goalColor: string,
   ancestors: string[],
+  goalHint: boolean,
 ) => {
+  if (goalHint === true) {
+    const res = await getGoalHints({
```

Comment:
why we are fetching hints here ?

Reply:
@Tushar-4781 The function getGoalHints() connects to the API mentioned in the issue when the user turns on the hints toggle and saves a goal. 
There's another function called getGoalHint() that fetches the hint value from the "hintsCollection" for a specific goal. To avoid confusion, we need better names for the getGoalHints() function. Any ideas? @tijlleenders

Reply:
- getHintsFromAPI()
- getHintsFromCollection()
? The days where we need to use short names to save space are over :)

Reply:
@vinaybadgujar102 I still don't understand properly why we are fetching hints here. I mean you are not using it anywhere so what's the point of fetching if?